server/channels/store/sqlstore/post_store.go

```diff
@@ -159,9 +159,11 @@ func (s *SqlPostStore) SaveMultiple(posts []*model.Post) ([]*model.Post, int, er
 		}
 		post.PreSave()
 		maxPostSize := s.GetMaxPostSize()
+
 		if err := post.IsValid(maxPostSize); err != nil {
 			return nil, idx, err
 		}
+		post.ValidateProps(rctx.Logger())
```

Comment:
I see the validation only done in save, save multiple, overwrite and overwrite multiple. I checked and the Update path, for example, doesn't seem to be passing through any of these four (I may have missed something).

Apart of that, we are doing the validation on the store layer. Theoretically, validation should be at a higher level (API or app levels). Nevertheless, I see we are calling post.IsValid just a line before this one, so I guess it is fine like this...

Reply:
We also validate the props when a post gets updated; see https://github.com/mattermost/mattermost/blob/b2f3ffb3980d89ca3d6ee72bda43886052bb00a6/server/channels/store/sqlstore/post_store.go#L357-L360

Reply:
I fully agree with your concern about validating on the store layer. That would indeed happen on higher levels. However, consistency is key. We are also doing other validation on the store layer, so let's keep that pattern. (And change it once and for all in a separate PR)